src/server_manager/electron_app/http/fetch.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as https from 'https';
+import {TLSSocket} from 'tls';
+import {urlToHttpOptions} from 'url';
+
+import type {IncomingMessage} from 'http';
+
+import type {HttpRequest, HttpResponse} from './types';
+
+export const fetchWithPin = async (
```

Comment:
NOTE: I remember having CORS issues with the DO api in the past. We could have a variant of this method without the fingerprint to call the DO APIs to avoid CORS issues.

Reply:
True!